src/faultlog/faultlog_main.cpp

```diff
@@ -143,46 +197,26 @@ int main(int argc, char** argv)
         // create fault log pel if there are service actions pending
         else if (createPel)
         {
-            int guardCount = GuardWithEidRecords::getCount(unresolvedRecords);
-            int manualGuardCount =
-                GuardWithoutEidRecords::getCount(unresolvedRecords);
-            int unresolvedPelsCount = UnresolvedPELs::getCount(bus);
-            int deconfigCount = DeconfigRecords::getCount();
-
-            if ((guardCount > 0) || (manualGuardCount > 0) ||
-                (unresolvedPelsCount > 0) || (deconfigCount > 0))
+            createNagPel(bus, unresolvedRecords);
+        }
+        // create bmc reboot pel
+        else if (bmcReboot)
+        {
+            // interested only in bmc reboot, host should have been in IPL
+            // runtime during bmc reboot
+            if (!isHostStateRunning(bus)) // host started
```

Comment:
We want to cover for bmc reboot case where in host should be already in running state so no need to wait for ipl state.  During re-ipl it will start host service again which will trigger nag dump